Source/Modules/r.cxx

```diff
@@ -2141,7 +2145,18 @@ int R::functionWrapper(Node *n) {
   if (destructor)
     Printv(f->code, "R_ClearExternalPtr(self);\n", NIL);
 
-  Printv(f->code, "return r_ans;\n}\n", NIL);
+  Printv(f->code, "return r_ans;\n", NIL);
+  
+  /* Error handling code */
+  Printv(f->code, "fail: SWIGUNUSED;\n", NIL);
+  if (need_cleanup) {
+    Printv(f->code, cleanup, NIL);
+  }
+  Printv(f->code, "  Rf_error(\"%s %s\", SWIG_ErrorType(SWIG_lasterror_code), SWIG_lasterror_msg);\n", NIL);
```

Comment:
Maybe I misunderstand the SWIG backend, but from comprehension, and taking the Python language as an example, the behavior of `SWIG_Error()`does not stop the execution of the program and continue to whatever follow. Instead this is the `SWIG_Fail;` macro that returns from the function (with a `goto fail;` statement). Both are used in the `SWIG_exception_fail()` function. I tried to recreate this for the `R` language. The `matlab` branch works also in the same way.

The use of `Rf_error()` instead of `Rf_warning()` is also to replicate the behavior of SWIG as it is within Python and Matlab. If SWIG fails somewhere in the code, then R should stop too with an error instead of displaying a warning and goes to the next instruction. Note: `Rf_warning()` can have the same behavior as `Rf_error()`(see the [interface / calling hierarchy](https://github.com/wch/r-source/blob/trunk/src/main/errors.c#L59)) but it depends on the kind of warning. 

@joequant may explain why he decided to use `Rf_warning()`.